src/lib.rs

```diff
@@ -83,6 +83,83 @@ enum Mode {
     WriteDisk { ownership: Ownership },
 }
 
+/// Get all files in a archive using `source` as a reader.
+/// # Example
+///
+/// ```no_run
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// use compress_tools::*;
+/// use std::fs::File;
+///
+/// let mut source = File::open("tree.tar")?;
+///
+/// let file_list = list_files(&mut source)?;
+/// # Ok(())
+/// # }
+/// ```
+pub fn get_files<R>(source: R) -> Result<Vec<String>>
+where
+    R: Read,
+{
+    run_with_archive(
+        Mode::AllFormat,
+        source,
+        |archive_reader, _, mut entry| unsafe {
+            let mut list_files = Vec::new();
+            loop {
+                match ffi::archive_read_next_header(archive_reader, &mut entry) {
+                    ffi::ARCHIVE_OK => {
+                        list_files.push(
+                            CStr::from_ptr(ffi::archive_entry_pathname(entry))
+                                .to_str()?
+                                .to_string(),
+                        );
+                    }
+                    _ => return Ok(list_files),
+                }
+            }
+        },
+    )
+}
+
+/// List all files in a archive using `source` as reader.
+///
+/// # Example
+///
+/// ```no_run
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// use compress_tools::*;
+/// use std::fs::File;
+///
+/// let mut source = File::open("tree.tar")?;
+///
+/// list_files(&mut source)?;
+/// # Ok(())
+/// # }
+/// ```
+pub fn list_files<R>(source: R) -> Result<()>
+where
+    R: Read,
+{
+    run_with_archive(
+        Mode::AllFormat,
+        source,
+        |archive_reader, _, mut entry| unsafe {
+            loop {
+                match ffi::archive_read_next_header(archive_reader, &mut entry) {
+                    ffi::ARCHIVE_OK => {
+                        println!(
+                            "{:#}",
+                            CStr::from_ptr(ffi::archive_entry_pathname(entry)).to_str()?
+                        );
+                    }
+                    _ => return Ok(()),
+                }
+            }
+        },
+    )
+}
```

Comment:
I think `list_files` is not needed. It can be written using `get_files` by library users if needed.

Reply:
I was think about if this function is really necessary and i agreed with you, can be removed because `get_files `can do the same thing.